### PR TITLE
Add ProduceLegacyHmacValues

### PR DESCRIPTION
**Pull request by danmosemsft**
Fix https://github.com/dotnet/corefx/issues/12364

@steveharter @bartonjs

### CODE REVIEW

File: src/System.Security.Cryptography.Algorithms/tests/HmacSha384Tests.cs [hunk 1]
```diff
@@ -21,6 +21,15 @@ protected override HashAlgorithm CreateHashAlgorithm()
         protected override int BlockSize { get { return 128; } }
 
         [Fact]
+        public void ProduceLegacyHmacValues()
+        {
+            var h = new HMACSHA384();
```
Comment (bartonjs) [inline src/System.Security.Cryptography.Algorithms/tests/HmacSha384Tests.cs]: Use a `using` statement, please :)

File: src/System.Security.Cryptography.Algorithms/tests/HmacSha512Tests.cs [hunk 2]
```diff
@@ -21,6 +21,15 @@ protected override HashAlgorithm CreateHashAlgorithm()
         protected override int BlockSize { get { return 128; } }
 
         [Fact]
+        public void ProduceLegacyHmacValues()
+        {
+            var h = new HMACSHA512();
```
Comment (bartonjs) [inline src/System.Security.Cryptography.Algorithms/tests/HmacSha512Tests.cs]: Ditto

**Comment by danmosemsft** [comment]
@dotnet-bot test this
@mmitche odd, I pushed an update ("CR feedback" above) but it didn't reset the CI.

**Comment by mmitche** [comment]
@danmosemsft Might have straddled the server shift (though I don't know why this would matter).  Let me know if you see this again.

File: src/System.Security.Cryptography.Algorithms/tests/HmacSha384Tests.cs [hunk 3]
```diff
@@ -20,6 +20,19 @@ protected override HashAlgorithm CreateHashAlgorithm()
 
         protected override int BlockSize { get { return 128; } }
 
+#if netcoreapp11
```
Comment (bartonjs) [inline src/System.Security.Cryptography.Algorithms/tests/HmacSha384Tests.cs]: Actually, I think the conditionals should be `netstandard17`... `netcoreapp11` is for API in .NET Core that isn't in .NET Standard.  (Applies to the other PRs as well)

**Pull request merged**